File: Retos/Reto #27 - CUENTA ATRÁS [Media]/typescript/magdielina.ts
/*
 * Crea una función que reciba dos parámetros para crear una cuenta atrás.
 * - El primero, representa el número en el que comienza la cuenta.
 * - El segundo, los segundos que tienen que transcurrir entre cada cuenta.
 * - Sólo se aceptan números enteros positivos.
 * - El programa finaliza al llegar a cero.
 * - Debes imprimir cada número de la cuenta atrás.
 */

const countdown = (start: number, interval: number, first : boolean = true) => {
    if(Number.isInteger(start) && Number.isInteger(interval) && (start > 0 || !first) && interval > 0) {
         if(start >= 0) {
            console.log(`Count... ${start}`);
            start--;
            setTimeout(countdown, interval * 1000, start, interval, false);

        }
     } else {
         console.log('Parameters are invalid, please provide only positive numbers');
     }
 }
 countdown(10, 1);

File: Retos/Reto #27 - CUENTA ATRÁS [Media]/java/Magdielina.java
public class Magdielina {
    public static void main(String[] args) {
        try {
            countdown(10,5);
        } catch (InterruptedException e) {
            System.out.println("Sorry!, there was an error during countdown.");
        }
    }

    private static void countdown(int start, int interval) throws InterruptedException{
        if (start > 0 && interval > 0) {
            for (; start >= 0; start--) {
                System.out.println("Count..." + start);
                Thread.sleep(interval * 1000);
            }
        } else {
            System.out.println("Please, provide only positive numbers.");
        }
    }
}